### PR TITLE
Add logic for module class

### LINKED ISSUE #41: Create Module class

### PR DESCRIPTION
**Pull request by HoJunHao2000**
* Able to add and delete modules
* Able to save and load modules
* Removed hardcoded modules

Resolves #41 and #42

### CODE REVIEW

**Comment by domoberzin** [comment]
I think to avoid conflict with tasks and deadlines we should suffix the Add/Remove commands with Module/Task/Deadline.
e.g. AddModuleCommand, AddTaskCommand. And for AddressBook we could have TaskBook and ModuleBook. If not merging all our branches may be quite tedious

File: src/main/java/modtrekt/logic/commands/AddCommand.java
Comment (domoberzin) [inline src/main/java/modtrekt/logic/commands/AddCommand.java:35]: Could perhaps rename to AddModuleCommand to avoid conflict with other add-based commands, same goes for removing, finding etc.

File: src/main/java/modtrekt/logic/commands/ListCommand.java
Comment (domoberzin) [inline src/main/java/modtrekt/logic/commands/ListCommand.java:19]: Nice job including the list command

File: src/main/java/modtrekt/ui/TaskCard.java
Comment (domoberzin) [inline src/main/java/modtrekt/ui/TaskCard.java:13]: Perhaps could add another class representing the Module card and Module panel, but I think could leave it to the people doing up the UI as well, but currently, this may intefere with the Task section when merging

File: src/main/java/modtrekt/logic/commands/AddCommand.java
Comment (domoberzin) [inline src/main/java/modtrekt/logic/commands/AddCommand.java:16]: iirc, our UG specifies that adding module should be done with `add -m <details>` or `add module <details>`, this will conflict with the task portion as it uses `add -t` for now

**Review by domoberzin — CHANGES_REQUESTED** [review]
Good job! Definitely was a hard task to modify the original code, good effort modifying the docs as well, just some minor things to change to make merging easier, and to follow our UG specifications. Nice work

File: README.md
Comment (jontmy) [inline README.md]: Was a global rename of AddressBook performed? This change looks unintentional.

File: src/main/java/modtrekt/ui/TaskCard.java
Comment (jontmy) [reply src/main/java/modtrekt/ui/TaskCard.java:13]: I've already created `ModuleCard` and `ModuleListPanel` in the `ui/modules` package in my PR - should this class be deleted?

File: src/main/java/modtrekt/ui/TaskListPanel.java
Comment (jontmy) [inline src/main/java/modtrekt/ui/TaskListPanel.java:11]: Likewise, I think this class is superseded by `ui/modules/ModuleListPanel`.

File: src/main/java/modtrekt/ui/modules/ModuleCard.java
Comment (jontmy) [inline src/main/java/modtrekt/ui/modules/ModuleCard.java:38]: Should the `get` methods be changed to instead return these attributes? Also, the use of `.value` in `getCredits()` causes the display text to be missing the `MCs` suffix.

Screenshot 2022-10-09 at 17 33 36

**Review by jontmy — COMMENTED** [review]
Great work! I had set the attributes of `Module` to be `public` instead of `private`  by accident - I'll fix this after this is merged.

**Review by vvidday — APPROVED** [review]
LGTM!

**Pull request merged**
